numpy/ma/core.py

```diff
@@ -3121,6 +3121,13 @@ def __getitem__(self, indx):
             if isinstance(indx, basestring):
                 if self._fill_value is not None:
                     dout._fill_value = self._fill_value[indx]
+                    # If we're indexing a multidimensional field in a
+                    # structured array (such as dtype("(2)i2,(2)i1")),
+                    # indexing fill_value will yield a multi-element
```

Comment:
Its a fun thing about how numpy treats field dtypes. Numpy is "smart" about it, it will assume that you actually do not want that dtype and just convert it to the appropriate array.

That may typically be good, but this PR is basically a symptom of there is seemingly being no way to make:
```
np.ones((), dtype='(3,)i')
```
to actually return what you would expect. Or even creating that array *at all*.